src/inmanta/execute/proxy.py

```diff
@@ -75,12 +75,20 @@ def _get_instance(self):
 
     @classmethod
     def unwrap(cls, item):
+        if item is None:
+            return NoneValue()
+
         if isinstance(item, DynamicProxy):
             return item._get_instance()
 
         if isinstance(item, list):
             return [cls.unwrap(x) for x in item]
 
+        if isinstance(item, dict):
+            if not all(isinstance(key, str) for key in item.keys()):
+                raise RuntimeException(None, "dict keys should be strings.")
```

Comment:
Indicate which value, with which type and dict value is violating.  (as much context as possible)